File: Try_GameEngine5/src/com/example/try_gameengine/action/visitor/MovementActionAttachToTargetSpriteVisitor.java
package com.example.try_gameengine.action.visitor;

import com.example.try_gameengine.action.MovementAction;
import com.example.try_gameengine.framework.Sprite;

/**
 * @author irons
 *
 */
public class MovementActionAttachToTargetSpriteVisitor implements IMovementActionVisitor{
	private Sprite sprite;
	
	/**
	 * MovementActionAttachToTargetSpriteVisitor
	 * @param sprite
	 */
	public MovementActionAttachToTargetSpriteVisitor(Sprite sprite) {
		this.sprite = sprite;
	}
	
	@Override
	public void visitComposite(MovementAction movementAction) {
		if(movementAction.getInfo()!=null){
			movementAction.getInfo().setSprite(sprite);
		}	
	}

	@Override
	public void visitLeaf(MovementAction movementAction) {
		if(movementAction.getInfo()!=null){
			movementAction.getInfo().setSprite(sprite);
		}
	}

}

File: Try_GameEngine5/src/com/example/try_gameengine/action/visitor/MovementActionAttachToTargetSpriteVisitor.kt
package com.example.try_gameengine.action.visitor

import com.example.try_gameengine.action.MovementAction
import com.example.try_gameengine.framework.Sprite

/**
 * @author irons
 // */
class MovementActionAttachToTargetSpriteVisitor(sprite: Sprite?) : IMovementActionVisitor {
    private val sprite: Sprite?

    /**
     * MovementActionAttachToTargetSpriteVisitor
     * @param sprite
     // */
    init {
        this.sprite = sprite
    }

    override fun visitComposite(movementAction: MovementAction) {
        if (movementAction.getInfo() != null) {
            movementAction.getInfo().setSprite(sprite)
        }
    }

    override fun visitLeaf(movementAction: MovementAction) {
        if (movementAction.getInfo() != null) {
            movementAction.getInfo().setSprite(sprite)
        }
    }
}
